airflow/contrib/operators/databricks_operator.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import time
+
+from airflow.exceptions import AirflowException
+from airflow.contrib.hooks.databricks_hook import DatabricksHook
+from airflow.models import BaseOperator
+
+POLL_SLEEP_PERIOD_SECONDS = 5
+LINE_BREAK = ("-" * 80)
+
+
+class DatabricksSubmitRunOperator(BaseOperator):
+    """
+    Submits an ephemeral run to Databricks.
+
+    https://docs.databricks.com/api/latest/jobs.html#runs-submit
+
+    Note that the named
+    parameters to this operator match the parameters exposed by
+    the Databricks ``api/2.0/jobs/runs/submit`` endpoint.
+
+    As a result, one way to instantiate a ``DatabricksSubmitRunOperator``
+    is to pass the same JSON object used to call ``api/2.0/jobs/runs/submit``
+    into the ``DatabricksSubmitRunOperator``. For example: ::
+
+        params = {
+          "new_cluster": {
+            "spark_version": "2.0.x-scala2.10",
+            "node_type_id": "r3.xlarge",
+            "aws_attributes": {
+              "availability": "ON_DEMAND"
+            },
+            "num_workers": 2
+          },
+          "libraries": [
+            {
+              "jar": "dbfs:/test.jar"
+            },
+            {
+              "maven": {
+                "coordinates": "org.jsoup:jsoup:1.7.2"
+              }
+            }
+          ],
+          "spark_jar_task": {
+            "main_class_name": "com.databricks.ComputeModels"
+          }
+        }
+        DatabricksSubmitRunOperator(task_id='spark_jar_run', **params)
```

Comment:
If we introduce new fields to the API call, this won't get picked up. Users should use `extra_api_parameters=params` instead.